src/common-components/search-bar/actions/index.js

```diff
@@ -0,0 +1,112 @@
+import {
+  knowledge_base,
+  random_data,
+  soft_skills_data,
+  domain_api,
+  language_api,
+  soft_skill_api
+} from '../../../common-services/api-endpoints';
+import { KNOWLEDGE_BASE, SOFT_SKILLS, RANDOM } from '../../../common-services/course_types';
+
+export const DELETE_SEARCH_RESULTS = 'delete-search-results';
+export const FETCH_SEARCH_RESULTS = 'fetch-search-results';
+export const NO_SEARCH_RESULTS = 'no-search-results';
+
+const soft_skill = 'soft_skill';
+const languages = 'languages';
+const domains = 'domains';
+
+
+export function fetchCourses(term) {
+  term = term.trim();
+  term = term.replace(/[^a-zA-Z0-9]/g, "-");
+
+  return function(dispatch) {
+
+    // chaining 'if else' for checking if term is present in 'domain', 'language', or 'softskills'
+    fetch(`${domain_api}?slug=${term}`)
+    .then(response => {
+      response.json()
+      .then(data => {
+        if(data.count) {
```

Comment:
what if something matches in the domain AS WELL AS in the language. We'll never find the language matches, right?

I think this should be done with a promise.all thing, so that we can merge together all search results.